telcell/celldb/pgdatabase.py

```diff
@@ -0,0 +1,414 @@
+            if distance_lower_limit_m is not None:
+                qwhere.append(
+                    f"NOT ST_DWithin(rd, 'SRID=4326;POINT({x} {y})', {distance_lower_limit_m})"
+                )
+        if date is not None:
+            qwhere.append("(date_start is NULL OR %s >= date_start)")
+            qwhere.append("(date_end is NULL OR %s < date_end)")
+            qargs.extend([date, date])
+
+        if radio is not None:
+            if isinstance(radio, str):
+                radio = [radio]
+            qwhere.append(f"({' OR '.join(['radio = %s'])})")
+            qargs.extend(radio)
+
+        if mcc is not None:
+            qwhere.append(f"mcc = {mcc}")
+        if mnc is not None:
+            qwhere.append(f"mnc = {mnc}")
+
+        if exclude is not None:
+            if isinstance(exclude, CellIdentity):
+                exclude = [exclude]
+            for addr in exclude:
+                add_qwhere, add_qargs = _build_cell_identity_query(addr)
+                qwhere.append(f"NOT ({add_qwhere})")
+                qargs.extend(add_qargs)
+
+        qorder = self._qorder
+        if random_order is not None and random_order:
+            qorder = "ORDER BY RANDOM()"
+        elif coords is not None:
+            x, y = point_to_rd(coords)
+            qorder = f"ORDER BY ST_Distance(rd, 'SRID=4326;POINT({x} {y})')"
+
+        count_limit = count_limit if count_limit is not None else self._count_limit
+
+        return PgCollection(
+            self._con, qwhere, qargs, qorder, count_limit, self._on_duplicate
+        )
+
+    def close(self):
+        if self._cur is not None:
+            self._cur.close()
+
+    def __enter__(self):
+        self._cur = self._con.cursor()
+        return self
+
+    def __exit__(self, type, value, tb):
+        self.close()
+
+    def __iter__(self):
+        if self._cur is None:
+            self._cur = self._con.cursor()
+
+        q = f"""
+            SELECT date_start, date_end, radio, mcc, mnc, lac, ci, eci, ST_X(rd), ST_Y(rd), azimuth
+            FROM antenna_light
+            WHERE {' AND '.join(qw for qw in self._qwhere)}
+            {self._qorder}
+        """
+        if self._count_limit is not None:
+            q += f" LIMIT {self._count_limit}"
+
+        self._cur.execute(q, self._qargs)
+        return self
+
+    def __next__(self):
+        row = self._cur.fetchone()
+        if row is None:
+            raise StopIteration
+
+        return _build_antenna(row)
+
+    def __len__(self):
+        with self._con.cursor() as cur:
+            cur.execute(
+                f"""
+                SELECT COUNT(*)
+                FROM antenna_light
+                WHERE {' AND '.join(qw for qw in self._qwhere)}
+            """,
+                self._qargs,
+            )
+
+            return cur.fetchone()[0]
+
+
+def csv_import(con, flo, progress: Callable = lambda x: x):
+    create_table(con)
+
+    reader = csv.reader(flo)
+    next(reader)  # skip header
```

Comment:
dit zou eventueel als dictreader kunnen, zodat je iets minder afhankelijk bent van als de volgorde in de csv misschien net iets anders zou zijn